### PR TITLE
pfsinitpak OK

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/libultra/io/pfsinitpak.c [hunk 1]
```diff
@@ -1,5 +1,126 @@
...
+    pfs->queue = queue;
+    pfs->channel = channel;
+    pfs->status = 0;
+
+    ret = __osPfsCheckRamArea(pfs);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osPfsSelectBank(pfs, 0);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:33]:
```suggestion
    ret = __osPfsSelectBank(pfs, PFS_ID_BANK_256K);
```

File: src/libultra/io/pfsinitpak.c [hunk 2]
```diff
@@ -1,5 +1,126 @@
...
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osContRamRead(pfs->queue, pfs->channel, PFS_ID_0AREA, temp);
+    if (ret != 0) {
+        return (ret);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:40]:
```suggestion
        return ret;
```

File: src/libultra/io/pfsinitpak.c [hunk 3]
```diff
@@ -1,5 +1,126 @@
...
+    id = (__OSPackId*)temp;
+    if ((id->checksum != sum) || (id->invertedChecksum != isum)) {
+        ret = __osCheckPackId(pfs, id);
+        if (ret != 0) {
+            pfs->status |= PFS_ID_BROKEN;
+            return ret;
+        }
+    }
+
+    if ((id->deviceid & 0x01) == 0) {
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:53]: Based on how this is assigned in `__osRepairPackId` I think this makes sense
```suggestion
    if (!(id->deviceid & 1)) {
```

File: src/libultra/io/pfsinitpak.c [hunk 4]
```diff
@@ -1,5 +1,126 @@
...
+        ret = __osRepairPackId(pfs, id, &newid);
+        if (ret != 0) {
+            if (ret == PFS_ERR_ID_FATAL) {
+                pfs->status |= PFS_ID_BROKEN;
+            }
+            return ret;
+        }
+        id = &newid;
+        if ((id->deviceid & 0x01) == 0) {
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:62]:
```suggestion
        if (!(id->deviceid & 1)) {
```

File: src/libultra/io/pfsinitpak.c [hunk 5]
```diff
@@ -1,5 +1,126 @@
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/pfsinitpak/osPfsInitPak.s")
+s32 __osPfsCheckRamArea(OSPfs* pfs);
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/pfsinitpak/__osPfsCheckRamArea.s")
+s32 osPfsInitPak(OSMesgQueue* queue, OSPfs* pfs, s32 channel) {
+    s32 ret;
+    u16 sum;
+    u16 isum;
+    u8 temp[BLOCKSIZE];
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:10]: Needs updating elsewhere too
```suggestion
    u8 buf[BLOCKSIZE];
```

File: src/libultra/io/pfsinitpak.c [hunk 6]
```diff
@@ -1,5 +1,126 @@
...
+    u8 temp1[BLOCKSIZE];
+    u8 temp2[BLOCKSIZE];
+    u8 saveReg[BLOCKSIZE];
+
+    ret = __osPfsSelectBank(pfs, PFS_ID_BANK_256K);
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osContRamRead(pfs->queue, pfs->channel, 0, saveReg);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:102]:
```suggestion
    ret = __osContRamRead(pfs->queue, pfs->channel, PFS_ID_0AREA, saveReg);
```

File: src/libultra/io/pfsinitpak.c [hunk 7]
```diff
@@ -1,5 +1,126 @@
...
+    if (ret != 0) {
+        return ret;
+    }
+
+    for (i = 0; i < BLOCKSIZE; i++) {
+        temp1[i] = i;
+    }
+
+    ret = __osContRamWrite(pfs->queue, pfs->channel, 0, temp1, 0);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:111]:
```suggestion
    ret = __osContRamWrite(pfs->queue, pfs->channel, PFS_ID_0AREA, temp1, 0);
```

File: src/libultra/io/pfsinitpak.c [hunk 8]
```diff
@@ -1,5 +1,126 @@
...
+    if (ret != 0) {
+        return ret;
+    }
+
+    ret = __osContRamRead(pfs->queue, pfs->channel, 0, temp2);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:116]:
```suggestion
    ret = __osContRamRead(pfs->queue, pfs->channel, PFS_ID_0AREA, temp2);
```

File: src/libultra/io/pfsinitpak.c [hunk 9]
```diff
@@ -1,5 +1,126 @@
...
+    if (ret != 0) {
+        return ret;
+    }
+
+    if (bcmp(temp1, temp2, BLOCKSIZE) != 0) {
+        return PFS_ERR_DEVICE;
+    }
+
+    return __osContRamWrite(pfs->queue, pfs->channel, 0, saveReg, 0);
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:125]:
```suggestion
    return __osContRamWrite(pfs->queue, pfs->channel, PFS_ID_0AREA, saveReg, 0);
```

File: src/libultra/io/pfsinitpak.c [hunk 10]
```diff
@@ -1,5 +1,126 @@
...
+    pfs->status |= PFS_INITIALIZED;
+
+    return ret;
+}
+
+s32 __osPfsCheckRamArea(OSPfs* pfs) {
+    s32 i = 0;
+    s32 ret = 0;
+    u8 temp1[BLOCKSIZE];
+    u8 temp2[BLOCKSIZE];
```
Comment (Thar0) [inline src/libultra/io/pfsinitpak.c:94]:
```suggestion
    u8 writeBuf[BLOCKSIZE];
    u8 readBuf[BLOCKSIZE];
```

**Review by Thar0 — CHANGES_REQUESTED** [review]

**Review by Thar0 — APPROVED** [review]

**Review by kyleburnette — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**